src/Adapt/Adapt.component.tsx

```diff
@@ -0,0 +1,101 @@
+// VENDOR
+import * as React from 'react';
+import classNames from 'classnames';
+import styled, {
+    up,
+    css,
+    down,
+    ThemeContext,
+} from '@xstyled/styled-components';
+import { layout, LayoutProps, space, SpaceProps } from '@xstyled/system';
+
+const { useContext } = React;
+
+interface ForEach {
+    [k: string]: any;
+}
+
+interface AdaptProps
+    extends React.HTMLAttributes<HTMLDivElement>,
+        LayoutProps,
+        SpaceProps {
+    className?: string;
+    forEach: ForEach;
+    children: (value: any, breakpoint: string) => any;
+}
+
+interface StyledParentProps
+    extends React.HTMLAttributes<HTMLDivElement>,
+        LayoutProps,
+        SpaceProps {
+    keys: string[];
+    className?: string;
+}
+
+const StyledParent = styled('div')<StyledParentProps>`
+  ${layout}
+  ${space}
+
+  ${({ keys }) =>
+      keys.map((current, index) => {
+          const next = keys[index + 1];
+          return css`
+              > :nth-child(${index + 1}) {
+                  ${next && up(next, css({ display: 'none' }))}
+                  ${current !== 'xs' && down(current, css({ display: 'none' }))}
+              }
+          `;
+      })}
+`;
+
+/**
+ * The purpose of Adapt is to display different content for different breakpoints.
+ * It accomplishes this via media queries and display: none. It expects a forEach
+ * prop object with values mapped to different breakpoints and children as a render
+ * prop. It then renders the child function for each breakpoint, passing in the
+ * associated value and the breakpoint as the first two arguments.
+ *
+ * Example usage:
+ * <Adapt forEach={{ xs: 'apple', md: 'banana' }}>
+ *   {(value, breakpoint) => (
+ *     <p>
+ *       {value}
+ *     </p>
+ *   )}
+ * </Adapt>
+ *
+ * =>
+ *   <p>apple</p>  // displays below md breakpoint
+ *   <p>banana</p>  // displays at and above md breakpoint
+ */
+export const Adapt = ({
+    className,
+    children,
+    forEach,
```

Comment:
`forEach` I'm also not completely sold on

Reply:
I'm not usually the name police, but `target` or something along those lines is better than using a protected JS namespace IMO.

Reply:
I avoided `for` and `each`, and while `forEach` is an array method, I wouldn't consider it protected namespace. The thing I like about it is the implication that the render prop (`children`) will render with every specified breakpoint.

```jsx
eachBreakpoint={{ ... }}
```
?

Reply:
What about `onBreakpoint`?

Reply:
Since the syntax is so similar to xstyled's utility, what about `breakpoints`? Keeps it consistent and the API is (basically) the same.

Reply:
Just gonna keep `forEach` :)